spec/amber/cli/commands/generator_spec.cr

```diff
@@ -6,32 +6,39 @@ include CLIHelper
 include CLIFixtures
 
 module Amber::CLI
-  begin
-    describe MainCommand::Generate do
-      context "scaffold" do
-        it "generates and compile generated app" do
-          ENV["AMBER_ENV"] = "test"
-          scaffold_app(TESTING_APP)
-          MainCommand.run ["generate", "scaffold", "Animal", "name:string"]
-          assert_app_compiled?(TEST_APP_NAME)
-          cleanup
-        end
+  describe MainCommand::Generate do
+    ENV["AMBER_ENV"] = "test"
+
+    context "when scaffolding an Amber app" do
+      scaffold_app(TESTING_APP)
+      MainCommand.run ["generate", "scaffold", "Animal", "name:string"]
```

Comment:
This line shouldn't be needed to test a `shards build` - did we just want something in there?

If we want a more complete test, it should be structured something like this:
```
context "scaffolding an Amber app"  
  assert that it builds
  test scaffold generator (in a describe block)
  test controller generator (in a describe block)
  test model generator (in a describe block)
  ...test each generator...
  assert that it still builds
  cleanup
end
```

Reply:
Practically speaking, we may not want two builds, until the shard downloading thing is optimized one way or another.

It might be better to test the shard building at the end - will it build after all the generators have been run, or did any of them introduce a compile time error?